### PR TITLE
Remove `Backend.characterisation`

### PR DESCRIPTION
**Pull request by cqc-alec**
After merging this I'll make a pytket pre-release and then update the extensions.

### CODE REVIEW

File: pytket/pytket/tailoring/__init__.py [hunk 1]
```diff
@@ -13,9 +13,8 @@
 # limitations under the License.
 
 """
-The circuit module provides an API to interact with the tket
-:py:class:`Characterisation` suite, with methods for noise shaping, characterisation and
-mitigation. This module is provided in binary form during the PyPI installation.
+The `tailoring` module provides an API to interact with tket's charcterisation suite,
```
Comment (sjdilkes) [inline pytket/pytket/tailoring/__init__.py:16]: Maybe this isn't the PR for it, but faithfully the `tailoring` module only provides an API for using frame randomisation methods, and maybe we should rename the whole module to frame randomisation in another PR?

If we save that for another review, it should be `tket's characterisation suite`.

**Review by sjdilkes — CHANGES_REQUESTED** [review]
One query.

File: pytket/pytket/tailoring/__init__.py (hunk 1 above)
Comment (cqc-alec) [reply pytket/pytket/tailoring/__init__.py]: Done. (Just me being lazy, I saw it was referring to the wrong module name and thought I'd correct it at the same time. But yes it should be a new PR.)

**Comment by ss2165** [comment]
> After merging this I'll make a pytket pre-release and then update the extensions.

The changes in #214 will also break things in extensions, should the fixes be sister PRs?

**Review by sjdilkes — APPROVED** [review]

**Comment by cqc-alec** [comment]
> > After merging this I'll make a pytket pre-release and then update the extensions.
> 
> The changes in #214 will also break things in extensions, should the fixes be sister PRs?

Good point. Maybe we should do a pre-release first without this change, fix those, and then do another pre-release with this change?

**Comment by cqc-alec** [comment]
> > > After merging this I'll make a pytket pre-release and then update the extensions.
> > 
> > 
> > The changes in #214 will also break things in extensions, should the fixes be sister PRs?
> 
> Good point. Maybe we should do a pre-release first without this change, fix those, and then do another pre-release with this change?

Actually, removing this doesn't break the extensions, so I'm going to merge and attempt another pre-release now.

(For some reason I'm having difficulty doing a pre-release. Seems the wrong version number is being applied by ertuptools_scm on some platforms...)

**Pull request merged**